### ISSUE TITLE
Are we thinking about an alias for if null and if not null?

### ISSUE THREAD

**Issue by jodinathan** [dart-lang/language#2332]
Sorry if already posted, I just couldn't find anything related.  

What about a shorthand to `if (foo == null)` and `if (foo != null)`?

Maybe some simple stuff like

`if (foo?) // same as foo == null` 

`if (foo!) // same as foo != null`

**Comment by mateusfccp** [comment]
I don't think this is valuable at all... It's not like we make null checks all the time, nor like it's a verbose expression, so...

**Comment by cedvdb** [comment]
There is the option of making null falsy. I'm personally not a fan of that but since you need ideas.

I'm also not a fan of adding 1 char tokens, I think they are obtuse when you don't use a language often and you think to yourself "what is this again ? Oh right". Obviously that's not the case with people reading this repository but it's something to keep in mind. 

> `foo ??= bar;`

I don't see the point of that either, it's barely used and `foo = foo ?? bar;` isn't much longer and less cryptic.

**Comment by jodinathan** [comment]
using `foo == null` gives me a feeling that I am checking the equality of a hardcode constant ie `bool isOdd(int n) => n == 1 || n == 3` instead of a `someInt.isOdd` or `n % 2 == 0` etc.  

As if the `null` keyword ever changes, it would break stuff.  

So adding a shorthand to nullability also opens room for other possibilities.

**Comment by cedvdb** [comment]
I think what you are saying is that it's not English (and I agree to some extent):

```
final employees = findCompanyEmployees();
final employeesCount = employees.length;

if (employeesCount.isOdd) {
   print('we have an odd number of employees');
}

final firstEmployeeName = employees.first.name;
if (firstEmployeeName != null) {  // this line is probably not understandable for someone who has never used a programing language
  print(firstEmployeeName);
}

// could be if (firstEmployeeName exists) {}
```

**Comment by munificent** [comment]
> Maybe some simple stuff like
> 
> `if (foo?) // same as foo == null`
> 
> `if (foo!) // same as foo != null`

Given how heavily overloaded `?` and `!` are already—nullable types, null-aware method call, conditional operator, null-aware spread, logical not—I think that would just cause more confusion than clarity. `== null` and `!= null` are already fairly short completely clear in their meaning, so I think it's unlikely we'd come up with something better.

**Comment by FMorschel** [comment]
I'd prefer to add some extensions to the language core:

```dart
extension NullableVar on Object? {
  bool get isNull => this == null;
  bool get isNotNull => this != null;
}

extension NullableString on String? {
  bool get isNullOrEmpty => this == null || this.isEmpty;
  bool get isNotNullOrEmpty => this != null && this.isNotEmpty;
}

extension NullableIterable on Iterable? {
  bool get isNullOrEmpty => this == null || this.isEmpty;
  bool get isNotNullOrEmpty => this != null && this.isNotEmpty;
}

extension NullableMap on Map? {
  bool get isNullOrEmpty => this == null || this.isEmpty;
  bool get isNotNullOrEmpty => this != null && this.isNotEmpty;
}
```

I'm not sure it would solve the "_shorthand_" part but it would feel more _Darty_ to have these getters IMO.
